sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -4630,4 +4630,144 @@ case class ArrayCompact(child: Expression)
     copy(child = newChild)
 }
 
+/**
```

Comment:
since we apply a NULL-handling approach different from existing array functions (array_remove, array_contains, array_position), would you mind document this difference in the comments?

Reply:
Done